domain/application/service/application.go

```diff
@@ -673,6 +679,23 @@ func makeCloudContainerArg(unitName coreunit.Name, cloudContainer CloudContainer
 	return result
 }
 
+// makeResourcesArgs creates a slice of AddApplicationResourceArg from ResolvedResources.
+// Returns an error if any of the ResolvedResources are invalid.
+func makeResourcesArgs(solvedResources ResolvedResources) ([]application.AddApplicationResourceArg, error) {
```

Comment:
solved?

Reply:
What do you mean ? Is in a suggestion to rename it `SolvedResources` or a question ?

ResolvedResources is borrowed to caller, where we have:

- Resources from charm (without any mention of revision)
- Resources overrides from command line (or anything which states that the actual resources isn't the last from the store, but some specific revision or an upload)

Resolved resources will be "all resources from charm, with a defined revision or having upload as origin".

Reply:
It's solved how?

Just called it resources.

Reply:
In my opinion, `ResolvedResources` is an ok name here, the only issue is the variable called `solvedResource`. Resolved and solved don't really make sense together, it'd be better just to call the parameter `resources`. Its clear that they are `ResolvedResources` from the type.

Reply:
Oh i get it now. Thanks !